toolchain/parser/parser_state.def

```diff
@@ -242,6 +242,29 @@ CARBON_PARSER_STATE(ExpressionLoopForBinary)
 //   1. ExpressionLoop
 CARBON_PARSER_STATE(ExpressionLoopForPrefix)
 
+// Completes an IfExpression.
```

Comment:
Please note the entry condition in `CARBON_PARSER_STATE(Expression)`, around line 187 (unfortunately can't comment there):

```
// If `If`:
//   1. Expression
//   2. ExpressionThen
//   3. ExpressionIfFinish
```

(at least as currently implemented)